src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -0,0 +1,314 @@
+        object_id = namespace.object_id or _get_object_id(graph_client, subscription=subscription)
+        namespace.filters = "Properties/ownerObjectId eq '{}'".format(object_id)
+    elif namespace.claimable:
+        namespace.filters = 'properties/allowClaim'
+    else:
+        namespace.filters = namespace.filters
+
+
+def _validate_location(namespace):
+    """
+    Selects the default location of the lab when location is not provided while creating vm.
+    """
+    if namespace.location is None:
+        lab_operation = devtestlabs_management_client.lab
+        lab = lab_operation.get_resource(namespace.resource_group, namespace.lab_name)
+        namespace.location = lab.location
+
+
+def _validate_expiration_date(namespace):
+    """
+    Validates expiration date if provided.
+    """
+    if namespace.expiration_date:
+        if datetime.datetime.utcnow().date() >= dateutil.parser.parse(namespace.expiration_date).date():
+            raise CLIError("Expiration date '{}' must be in future.".format(namespace.expiration_date))
+
+
+def _validate_network_parameters(namespace):
+    """
+    Selects the DevTest Lab's virtual network and subnet if not provided and updates namespace
+    """
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    vnet_operation = devtestlabs_management_client.virtual_network
+
+    if not namespace.vnet_name:
+        lab_vnets = list(vnet_operation.list(namespace.resource_group, namespace.lab_name, top=1))
+        if not lab_vnets:
+            err = "Unable to find any virtual network in the '{}' lab.".format(namespace.lab_name)
+            raise CLIError(err)
+        else:
+            lab_vnet = lab_vnets[0]
+            namespace.vnet_name = lab_vnet.name
+            namespace.lab_virtual_network_id = lab_vnet.id
+    else:
+        namespace.lab_virtual_network_id = resource_id(subscription=get_subscription_id(),
+                                                       resource_group=namespace.resource_group,
+                                                       namespace='microsoft.devtestlab',
+                                                       type='labs',
+                                                       name=namespace.lab_name,
+                                                       child_type='virtualnetworks',
+                                                       child_name=namespace.vnet_name)
+
+    # Select default subnet for selected vnet when subnet is not provided
+    if not namespace.subnet:
+        # Get the first subnet of the lab's virtual network
+        lab_vnet = vnet_operation.get_resource(namespace.resource_group, namespace.lab_name, namespace.vnet_name)
+        namespace.subnet = lab_vnet.subnet_overrides[0].lab_subnet_name
+
+        # Determine value for disallow_public_ip_address based on subnet's use_public_ip_address_permission property
+        if lab_vnet.subnet_overrides[0].use_public_ip_address_permission == 'Allow':
```

Comment:
now that you have a more updated code, you should also use the sharedPublicIpAddressConfiguration to set this value. If it has any entries, then public ip address defaults to false, even if use public ip is true.

In addition, if it has entries, then we also set that value on the vm with the correct port based on the os type.